### PR TITLE
13355 removed proxy* methods from RouteSetup

### PR DESCRIPTION
**Pull request by chunlokling-da**
Fixes part of #13355
Refactor `PackagesAndDars`, `MeteringReportEndpoint`, `Parties` and `UserManagement`

### CODE REVIEW

**Comment by chunlokling-da** [comment]
`toRoute2` will be renamed to `toRoute` eventually

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala [hunk 1]
```diff
@@ -109,6 +110,111 @@ class Endpoints(
...
+          packageManagementService.uploadDarFile(
+            jwt,
+            toLedgerId(payload.ledgerId),
+            source.mapMaterializedValue(_ => NotUsed),
+          )
+        )
+      ): ET[Unit]
+    } yield domain.OkResponse(())
+    responseToRoute(httpResponse(res))
+  }
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala:188]: It's more important that the route implementation remain separated than that the `routeSetup` argument is removed. So this should move back to where it was for this PR.

**Review by S11001001 — APPROVED** [review]
_Very_ nice job with the abstraction. Please move the case mentioned back to `PackagesAndDars` and we're good to go. 👍

**Review by ray-roestenburg-da — APPROVED** [review]
LGTM and as @S11001001 said.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/RouteSetup.scala [hunk 2]
```diff
@@ -147,10 +119,23 @@ private[http] final class RouteSetup(
...
+        withJwtPayload[Source[ByteString, Any], JwtPayloadLedgerIdOnly](
+          (j, req.entity.dataBytes)
+        ).run
+    }
+
   private[this] def data(entity: RequestEntity): Future[String] =
     entity.toStrict(maxTimeToCollectRequest).map(_.data.utf8String)
 
-  private[this] def inputJsVal(req: HttpRequest)(implicit
+  private[http] def inputJsVal(req: HttpRequest)(implicit
```
[the remark this answers is not in the document]
Comment (chunlokling-da) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/RouteSetup.scala:138]: https://github.com/digital-asset/daml/issues/15860

**Pull request merged**